File: Pygame @westboy31/TileMapGenerator.py
#Autor : westboy31
#Tile Map Generator : https://westboy.itch.io/tilemapgenerator
import os, json, pygame

Map = {}
Map["images"] = {}
Map["tiles"] = {}
Map["items"] = {}
Map["collisions"] = []
tmg_json = {}

def loadFiles(jsonPath) :
    global Map
    tmg_jsonFile = open(jsonPath)
    tmg_folder = os.path.dirname(jsonPath) + "/"

    global tmg_json
    tmg_json = json.load(tmg_jsonFile)
    tmg_TilesList = list(dict.fromkeys(tmg_json["TilesList"]))

    for img in tmg_TilesList :
        tile = pygame.image.load(tmg_folder + img)
        Map["images"][img.split(".")[0]] = tile

    for img in tmg_json["ItemsList"] :
        item =  pygame.image.load(tmg_folder + img)
        Map["images"][img.split(".")[0]] = item

    return Map["images"]


def drawMap(screen) :

    for T in list(tmg_json["Tiles"]) :
        tile = Map["images"][T["Tile"]]
        tile = pygame.transform.flip(tile, T["FlipHorizontally"] , T["FlipVertically"] )
        tile = screen.blit(tile , ( T["X"] * tmg_json["TileSize"] , T["Y"] * tmg_json["TileSize"] ) )
        Map["tiles"][  "ix"+ str(T["X"]) + "iy" + str(T["Y"]) ] = tile

    for I in list(tmg_json["Items"]) :
        item = screen.blit(Map["images"][I["Item"].split(".")[0]], ( I["X"]  , I["Y"] ) )
        Map["items"][  "x"+ str(I["X"]) + "y" + str(I["Y"]) ] = item

    if tmg_json["Map"] == "PlatformerHorizontal" or tmg_json["Map"] == "PlatformerMultidirectional" :
        for C in list(tmg_json["Collisions"]) :
            collision = pygame.Rect( C["X"], C["Y"], C["Width"], C["Height"] )
            Map["collisions"].append(collision)

    return Map

File: Love2D @westboy31/TileMapGenerator.lua
--Author : westboy31
--Tile Map Generator : https://westboy.itch.io/tilemapgenerator

json = require("JSON")
print(json)

local tmg = {}

local TilesList = {}
local tmp_img = ""
local images = {}
local map 
local tiles = {}
local orientationX = 1
local orientationY = 1

function tmg.loadFiles( jsonPath )
	local j = love.filesystem.read(jsonPath)
	map = json:decode(j)
	
	for i_image,img in pairs(map["TilesList"]) do
		if  tmp_img ~= img then
			table.insert(TilesList, 1, img)
			tmp_img = img
		end
	end

	for i,img in ipairs(TilesList) do
		local stop = string.find(img,'.',1,true) 
		local imgName = string.sub(img, 1, stop-1)
		images[imgName] = love.graphics.newImage("assets/" .. img)
	end

	for i,img in ipairs(map["ItemsList"]) do
		local stop = string.find(img,'.',1,true) 
		local imgName = string.sub(img, 1, stop-1)
		images[imgName] = love.graphics.newImage("assets/" .. img)
	end


return images
end	


function tmg.draw()
	for t,T in pairs(map["Tiles"]) do
		if T.FlipHorizontally then
	    	orientationX = -1
		else 
			orientationX = 1
		end

		if T.FlipVertically then
	    	orientationY = -1
		else 
			orientationY = 1
		end
	
		love.graphics.draw(images[T.Tile], 
					   	   T.X * map["TileSize"] + map["TileSize"]/2 ,
					   	   T.Y * map["TileSize"] + map["TileSize"]/2 ,
					       0 , orientationX , orientationY ,
					       map["TileSize"]/2 , map["TileSize"]/2 )

	end


	for i,I in pairs(map["Items"]) do
		local stop = string.find(I.Item,'.',1,true) 
		local itemName = string.sub(I.Item, 1, stop-1)
	    love.graphics.draw(images[itemName], I.X , I.Y )

	end


end



return tmg
